src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5391,6 +5391,15 @@ private BoundExpression MakeBadExpressionForObjectCreation(SyntaxNode node, Type
                         break;
                     }
 
+                case BoundKind.ImplicitIndexerAccess:
+                    var implicitIndexer = (BoundImplicitIndexerAccess)boundMember;
+                    MessageID.IDS_ImplicitIndexInitializer.CheckFeatureAvailability(diagnostics, implicitIndexer.Syntax);
+
+                    hasErrors |= isRhsNestedInitializer
+                        && !CheckNestedObjectInitializerPropertySymbol(GetPropertySymbol(implicitIndexer, out _, out _), leftSyntax, diagnostics, hasErrors, ref resultKind);
```

Comment:
>GetPropertySymbol(implicitIndexer, out _, out _) [](http://example.com/codeflow?start=71&length=48)

Are we going to get here for an array or a slice case? It looks like this helper is going to return null then.